filter_test.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"testing"
+
+	"github.com/miekg/dns"
+	"gopkg.in/yaml.v2"
+)
+
+func TestUnmarshalFilter(t *testing.T) {
+	var filter nameFilter
+	var yamlBytes = []byte(`[ sie-network.net, dnsdb.info ]`)
+
+	err := yaml.Unmarshal(yamlBytes, &filter)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func testLookup(f nameFilter, name string, res bool) func(t *testing.T) {
+	return func(t *testing.T) {
+		msg := &dns.Msg{Question: []dns.Question{dns.Question{Name: name}}}
```

Comment:
Nitpick, but this can be simplified to `[]dns.Question{{Name: name}}}`

Reply:
Fixed/simplified.